environment-variables/validate-environment-variables/snippet.go

```diff
@@ -0,0 +1,28 @@
+package snippets
+
+import (
+	"github.com/kelseyhightower/envconfig"
+)
+
+// See https://godoc.org/github.com/kelseyhightower/envconfig
+type Specification struct {
+	Regular              string
+	Array                []int
+	Map                  map[string]int
+	SplitWithUnderscores int `split_words:"true"`
+	WithDefaultValue     int `default:"42"`
+	OverriddenName       int `envconfig:"MANUAL_OVERRIDE"`
+	EmbeddedStruct       EmbeddedSpecification
+	Required             int `required:"true"`
+}
+
+type EmbeddedSpecification struct {
+	Value int
+}
+
+// Reads and validates environment variables according to the Specification struct
+func validateEnvironmentVariables(namePrefix string) (Specification, error) {
```

Comment:
the Go style is to be terse when the communication is going to mutually understood (e.g., by convention, or because of proximity to use) -- in this case, it's a bit more idiomatic to say `EnvVars` not `EnvironmentVariables`.  

Read through this: https://dave.cheney.net/practical-go/presentations/gophercon-singapore-2019.html#_identifiers

Reply:
:+1: